code-experiments/src/logger_bbob.c

```diff
@@ -447,31 +453,53 @@ static void logger_bbob_evaluate(coco_problem_t *problem, double *x, double *y)
       if (initial_solution_fvalue < y[0]) {
 			
         for (i = 0; i < problem->number_of_variables; i++) {
-          x[i] = inner_problem->initial_solution[i];
+          z[i] = inner_problem->initial_solution[i];
           logger->best_solution[i] = inner_problem->initial_solution[i];   
         }
-        y[0] = initial_solution_fvalue; 
+        z_objvalue = initial_solution_fvalue; 
       }
       else {
-        for (i = 0; i < problem->number_of_variables; i++)
+        for (i = 0; i < problem->number_of_variables; i++) {
           logger->best_solution[i] = x[i]; 
+          z[i] = x[i];
+        }
+        z_objvalue = y[0];
       }
     }
-    logger->best_fvalue = y[0];
-    is_feasible = 1;
+    logger->best_fvalue = z_objvalue;
   }
-  else if ((logger->number_of_evaluations == 1) || (y[0] < logger->best_fvalue && is_feasible)){
+  else if ((logger->number_of_evaluations == 1) || (y[0] < logger->best_fvalue && is_feasible)) {
     logger->best_fvalue = y[0];
-    for (i = 0; i < problem->number_of_variables; i++)
+    for (i = 0; i < problem->number_of_variables; i++) {
       logger->best_solution[i] = x[i];
+      z[i] = x[i];
+    }
+    z_objvalue = y[0];
+  }
+  else {
+	  /* No logging if not feasible or not better than best. */
+		coco_free_memory(z);
+		return;
+  }
+
+	printf("z_objvalue = %e\n", z_objvalue);
```

Comment:
Are you sure these `print`s were meant to stay?

Reply:
Ups. Sorry. Many thanks for catching this. The current version shouldn't have these lines anymore.